presto-main/src/main/java/io/prestosql/server/protocol/StatementResource.java

```diff
@@ -220,7 +220,7 @@ private static Response toResponse(Query query, QueryResults queryResults)
 
         // add set session properties
         query.getSetSessionProperties().entrySet()
-                .forEach(entry -> response.header(PRESTO_SET_SESSION, entry.getKey() + '=' + entry.getValue()));
+                .forEach(entry -> response.header(PRESTO_SET_SESSION, entry.getKey() + '=' + urlEncode(entry.getValue())));
```

Comment:
urlEncode may produce any 8-bit character, especially for UTF-8 data. This is not safe for headers, which are required to be:

```
TEXT           = <any OCTET except CTLs, but including LWS>
CTL            = <any US-ASCII control character (octets 0 - 31) and DEL (127)>
```

Let's use base64, instead.

Reply:
I misspoke. Url-encoded data only contains ASCII (any non-ASCII gets percent-encoded). However, I haven't looked at whether the reserved characters in url encoding overlaps with reserved characters in headers, which would still cause a problem.

Reply:
Good observation. This can output `a-z A-Z 0-9 + - _ . * %`. Is any of these unsafe?

Reply:
BTW we already use urlEncode for prepared statement and roles.

Reply:
Yeah, I all those should be safe. Also, since we use it for prepared statements, it’s good evidence there’s no issue, so disregard my earlier comment about base64.